jms/src/main/scala/akka/stream/alpakka/jms/scaladsl/JmsProducer.scala

```diff
@@ -15,8 +15,10 @@ object JmsProducer {
   /**
    * Scala API: Creates an [[JmsProducer]] for [[JmsMessage]]s
    */
-  def flow[T <: JmsMessage](settings: JmsProducerSettings): Flow[T, T, NotUsed] =
+  def flow[T <: JmsMessage](settings: JmsProducerSettings): Flow[T, T, NotUsed] = {
+    require(settings.destination.isDefined, "Producer destination must be defined in regular producer flow")
```

Comment:
Great to check early on.

Reply:
I was also thinking of introducing an intermediate `JmsIncompleteProducerSettings` type that gets the destination methods `toQueue`, `toTopic` etc, and that those methods yield a `JmsProducerSettings` instance - similar to the `JmsMessage` and `JmsDirectedMessage` types that were previously in this PR. Would break compatibility, but maybe something worthwhile for the 1.0 release?